R/nse-defuse.R

```diff
@@ -491,7 +491,10 @@ endots <- function(call,
   })
 
   if (splice_dots) {
-    dots <- flatten_if(dots, is_spliced)
+    with_options(
+      lifecycle_verbosity = "quiet",
+      dots <- flatten_if(dots, is_spliced)
+    )
```

Comment:
I'm somewhat worried about the performance impact of this in `enquos()`

Historically we've had performance issues with lifecycle warnings in performance sensitive functions, and `with_options()` also has non negligible overhead in setting up exit handlers

Reply:
On the other hand endots is already quite slow.

Reply:
You're right the overhead is bad, I'll change the implementation to something else.

Reply:
I changed the implementation here: https://github.com/r-lib/rlang/commit/2fa538f8bedb58c82293f82f50e620dedd6aef7e

Before:

```r
fn <- function(a, ..., b) enquos(a, ..., b)

bench::mark(
  fn(foo, bar, b = baz),
  iterations = 10000
)
#> # A tibble: 1 × 13
#>   expression                 min median itr/s…¹ mem_a…² gc/se…³ n_itr
#>   <bch:expr>            <bch:tm> <bch:>   <dbl> <bch:b>   <dbl> <int>
#> 1 fn(foo, bar, b = baz)   20.7µs 22.4µs  38632.      0B    7.73  9998
```

After:

```r
#> # A tibble: 1 × 13
#>   expression                 min median itr/s…¹ mem_a…² gc/se…³ n_itr
#>   <bch:expr>            <bch:tm> <bch:>   <dbl> <bch:b>   <dbl> <int>
#> 1 fn(foo, bar, b = baz)   23.6µs   26µs  36994.      0B    11.1  9997
```

The deprecation put this example in the half-ms range.

Reply:
Yea that checks out, I remember the lifecycle infrastructure being around that slow

And yea even `with_options()` has a chunk of overhead

``` r
bench::mark(with_options(lifecycle_verbosity = "quiet", 1))
#> # A tibble: 1 × 6
#>   expression                                          min  median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                                     <bch:tm> <bch:t>     <dbl> <bch:byt>    <dbl>
#> 1 with_options(lifecycle_verbosity = "quiet", 1)   92.6µs   103µs     9206.      688B     12.7
```